recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -142,6 +142,16 @@ this.PreferenceExperiments = {
       previousPreferenceValue: Preferences.get(preferenceName, undefined),
     };
 
+    if (experiment.previousPreferenceValue) {
+      const previousType = typeof experiment.previousPreferenceValue;
+      const givenType = typeof preferenceValue;
```

Comment:
Instead of using `typeof`, there's an API for getting the type of a preference that you'll want to use: [nsiPrefBranch.getPrefType](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIPrefBranch#getPrefType()).

That interface is available on `Services.prefs`. `RecipeRunner.jsm` is an example of how to import and use that object.